runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcGetDataStream.java

```diff
@@ -504,9 +504,6 @@ private void queueRequestAndWait(QueuedRequest request)
       trySendBatch(batch);
```

Comment:
should we also remove the prevBatch waiting and instead just try to send?  It means we are contending on the synchronized block and may not maintain the ordering but the synchronized primitive may handle that better than our daisy-chaining of waitForSendOrFailNotification across threads.

Tests would likely need updating if they rely on the ordering. But would be interesting to see if that also helps out on the profile.

Reply:
+1 removing it should be fine. Will double check and send it as a separate change.